models/Form.php

```diff
@@ -0,0 +1,111 @@
+<?php
+// Permet d'avoir accès à la base de données
+require_once "Model.php";
+
+// Cette classe sert à manipuler tout ce qui concerne le formulaire de contact de la page d'accueil
+class Form extends Model
+{
+    //Formulaire d'accueil
+    function form_page_home_cv(){
+      if(filter_has_var(INPUT_POST, 'submit')){
+
+                $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name', FILTER_SANITIZE_STRING)));
+                $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_SANITIZE_EMAIL, FILTER_VALIDATE_EMAIL)));
+                $subject = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'subject', FILTER_SANITIZE_STRING)));
+                $message = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'message', FILTER_SANITIZE_STRING)));
+
+                
+                $errors = [];
+
+                if(empty($name) || empty($email) || empty($subject) || empty($message)){
+                    $errors['empty'] = "Veuillez remplier tous les champs";
+                }
+                if(strlen($name) < 5){
+                    $errors['name'] = "Votre nom doit contenir au moins 5 caractères.";
+                }
+                if(strlen($subject) < 5){
+                    $errors['subject'] = "Votre sujet doit contenir au moins 5 caractères.";
+                }
+                if(strlen($message) < 5){
+                    $errors['message'] = "Votre message doit contenir au moins 50 caractères.";
+                }
+                
+                if(!empty($errors)){
+                    ?>
+                        <div class="card red">
+                            <div class="card-content white-text">
+                                <?php
+                                foreach($errors as $error){
+                                    echo $error."</br>";
+                                }
+                                ?>
+                            </div>
+                        </div>
+                    <?php
+                }else{
+                    Form::contact_mail($name,$email,$subject,$message);
+                    Form::contact_mail_user($name,$email,$subject,$message);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'Form' in method 'form_page_home_cv'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5349276)